### PR TITLE
lxqt-base/lxqt-config : fix building against >=kde-plasma/kscreen-5.26.90

### PR DESCRIPTION
**Pull request by V3n3RiX**
Bug: https://bugs.gentoo.org/894468
See: https://github.com/lxqt/lxqt-config/issues/903
See: https://github.com/lxqt/lxqt-build-tools/pull/83
See: https://github.com/lxqt/lxqt-config/pull/915

I tested and works fine on my system. Ideally, all LXQT stack would need a rebuild after the cpp standard change to lxqt-build-tools

### CODE REVIEW

File: lxqt-base/lxqt-config/files/kscreen-52690.patch [hunk 1]
```diff
@@ -0,0 +1,80 @@
+From 6add4e4f0040693e7c4242fbae48c9d32007686c Mon Sep 17 00:00:00 2001
```
Comment (thesamesam) [inline lxqt-base/lxqt-config/files/kscreen-52690.patch:1]: Please include a link to the patch source upstream/wherever you got it from.

File: dev-util/lxqt-build-tools/files/cpp17-standard.patch [hunk 2]
```diff
@@ -0,0 +1,27 @@
+From 4a3a7038b1927dc43942a930db103023efdcbe2c Mon Sep 17 00:00:00 2001
```
Comment (thesamesam) [inline dev-util/lxqt-build-tools/files/cpp17-standard.patch:1]: Please include any relevant references, including the patch source, and any relevant Gentoo bugs at the top of the patch.

File: lxqt-base/lxqt-config/lxqt-config-1.2.0-r1.ebuild [hunk 3]
```diff
@@ -49,6 +49,14 @@ DEPEND="
 "
 RDEPEND="${DEPEND}"
 
+src_prepare() {
+	if has_version ">=kde-plasma/libkscreen-5.26.90" ; then
```
Comment (thesamesam) [inline lxqt-base/lxqt-config/lxqt-config-1.2.0-r1.ebuild:53]: We don't like to do this as it's brittle. If it's purely about fixing an implicit include, we should be able to do this unconditionally?

File: lxqt-base/lxqt-config/files/kscreen-52690.patch (hunk 1 above)
Comment (V3n3RiX) [reply lxqt-base/lxqt-config/files/kscreen-52690.patch:1]: done

File: dev-util/lxqt-build-tools/files/cpp17-standard.patch (hunk 2 above)
Comment (V3n3RiX) [reply dev-util/lxqt-build-tools/files/cpp17-standard.patch:1]: done

File: lxqt-base/lxqt-config/lxqt-config-1.2.0-r1.ebuild (hunk 3 above)
Comment (V3n3RiX) [reply lxqt-base/lxqt-config/lxqt-config-1.2.0-r1.ebuild:55]: I agree, it is a bit brittle, however I'm not so sure the changes would work with an old kscreen. I will test and if it does, I'll make it unconditional.

**Comment by V3n3RiX** [comment]
I built all the lxqt stack with and without the cpp17-standard.patch against lxqt-build-tools. Also, as you suspected, the kscreen.patch against lxqt-config can be applied unconditionally, it built just fine against both kscreen-5.26.5 and kscreen-5.27.0. I pushed a change to make it unconditional.

**Comment by Chiitoo** [comment]
Was hoping there'd be point releases somewhat soon, but well, I didn't do anything to fix this when I noticed it already ages ago... aside from that issue opening, so can blame myself for that!

I need some time to test things out, hopefully tomorrow will be better for this, but some quick thoughts on things I noticed:

- The commits are missing 'Signed-off-by: Name <e-mail>' (see: https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- The 1.2.0 ebuilds should be left alone for now, especially since they're stable keyworded, and the new revisions be pushed with unstable keywords (probably should stabilise them pretty soon though, before, or at the latest with plasma-5.27.0 (this is not yet an issue with 5.26.5, right?).
- The commits should be squashed, so that there are no fix-up commits like the 'apply kscreen patch unconditionally'.  You can 'git rebase -i HEAD~4' or so and adjust the commits, and then force push to the branch used for this PR to keep the commits clean.
- The patch files could use the program name in their filenames (see: https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/).

Thanks!

**Comment by V3n3RiX** [comment]
> Was hoping there'd be point releases somewhat soon, but well, I didn't do anything to fix this when I noticed it already ages ago... aside from that issue opening, so can blame myself for that!
> 
> I need some time to test things out, hopefully tomorrow will be better for this, but some quick thoughts on things I noticed:
> 
>     * The commits are missing 'Signed-off-by: Name ' (see: https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
> 
>     * The 1.2.0 ebuilds should be left alone for now, especially since they're stable keyworded, and the new revisions be pushed with unstable keywords (probably should stabilise them pretty soon though, before, or at the latest with plasma-5.27.0 (this is not yet an issue with 5.26.5, right?).
> 
>     * The commits should be squashed, so that there are no fix-up commits like the 'apply kscreen patch unconditionally'.  You can 'git rebase -i HEAD~4' or so and adjust the commits, and then force push to the branch used for this PR to keep the commits clean.
> 
>     * The patch files could use the program name in their filenames (see: https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/).
> 
>     * Also, not sure that anything will complain about this, but usually we have something like 'dev-util/lxqt-build-tools: commit message' so that there is no space after the package name like so: 'dev-util/lxqt-build-tools : commit message''
> 
> 
> Thanks!

All done! Thank you for your review!

**Review by Chiitoo — COMMENTED** [review]
Thanks!

Forgot to mention that the Gentoo bug should also be mentioned in the commit message, which I've done now while also amending a few other things to get the fixes in now rather than another day:

- Added links to the upstream source and Gentoo bug in the patches themselves as well.
- The commit messages wer all a one line with the signed-off tag, though not shown like that on GitHub for some reason.  I added some line breaks there, and shortened the C++17 message a bit (related: https://devmanual.gentoo.org/ebuild-maintenance/git/#git-commit-message-format).
- Updated the 'Copyright 1999-2022 Gentoo Authors' line (copyright year).  Using 'pkgdev commit' from 'dev-util/pkgdev' does this automatically.  :]
- Added links to the upstream commits to the commit messages.

**Pull request closed without merging**